frontend/dockerfile/instructions/parse.go

```diff
@@ -84,6 +92,11 @@ func ParseInstruction(node *parser.Node) (v interface{}, err error) {
 	case command.Copy:
 		return parseCopy(req)
 	case command.From:
+		if !linter.ValidateStageNameCasing(req.args) {
+			location := linter.FirstNodeRange(node)
```

Comment:
Not sure why there is the extra function to get the first range instead of just returning all.

Reply:
`opt.Warn` doesn't take a list of ranges, it takes a single range. I figured grabbing the first range was sensible in that case.

Reply:
Looks like the array goes missing in conversion in https://github.com/moby/buildkit/blob/master/frontend/dockerfile/builder/build.go#L244-L253 . The actual type in API is array.

Reply:
Cool, will fix that. 👍